pkg/shortnames/shortnames.go

```diff
@@ -0,0 +1,154 @@
+package shortnames
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
+	"github.com/containers/image/v5/types"
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// The default registry to use in case no unqualified-search registries are
+// specified.
+const defaultRegistry = "docker.io"
+
+// IsShortName returns true if the specified input is a "short name" and an
+// error that may occur when the input could not be parsed.  A "short name"
+// refers to a container image without a fully-qualified reference, and is
+// hence missing a registry (or domain).
+//
+// Examples:
+//  * short names: "image:tag", "library/fedora"
+//  * not short names: "quay.io/image", "localhost/image:tag", "server.org:5000/lib/image"
+func IsShortName(input string) (bool, error) {
+	repo, err := reference.Parse(input)
+	if err != nil {
+		return false, errors.Wrapf(err, "cannot parse input: %q", input)
+	}
+	named, ok := repo.(reference.Named)
+	if !ok {
+		return false, nil
+	}
+
+	registry := reference.Domain(named)
+	if strings.ContainsAny(registry, ".:") || registry == "localhost" {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func completeNameWithRegistry(name, registry string) string {
```

Comment:
- Return a `reference.Named` (which may include a tag/digest)
- Ideally the input should be _only_ a value already processed by the “is the input a short name” question. Compare `parts.go`. Strings Are Bad™️.